kore/src/Kore/Logger/Output.hs

```diff
@@ -113,25 +132,117 @@ data KoreLogOptions = KoreLogOptions
     -- ^ extra entries to show, ignoring 'logLevel'
     , debugAppliedRuleOptions :: DebugAppliedRuleOptions
     , debugAxiomEvaluationOptions :: DebugAxiomEvaluationOptions
+    , debugSolverOptions :: DebugSolverOptions
     }
     deriving (Eq, Show)
 
 -- | Internal type used to add timestamps to a 'LogMessage'.
 data WithTimestamp = WithTimestamp SomeEntry LocalTime
 
+data LoggerStartData =
+    LoggerStartData
+        { logType :: !KoreLogType
+        , ioMode :: !IOMode
+        , handleLogger :: !(Handle -> LogAction IO SomeEntry)
+        }
+
 -- | Generates an appropriate logger for the given 'KoreLogOptions'. It uses
 -- the CPS style because some outputters require cleanup (e.g. files).
 withLogger
-    :: Trans.MonadIO m
-    => KoreLogOptions
-    -> (LogAction m SomeEntry -> IO a)
+    :: KoreLogOptions
+    -> (LogAction IO SomeEntry -> IO a)
     -> IO a
-withLogger koreLogOptions@KoreLogOptions { logType } continue =
+withLogger
+    koreLogOptions@KoreLogOptions
+        { logType
+        , debugSolverOptions = DebugSolverOptions {logFile = smtSolverLogFile}
+        }
+    continue
+  =
+    startLoggers
+        ( mainLoggerStartData :|  [ smtSolverLoggerStartData ])
+        []
+        continue
+  where
+    mainLoggerStartData = LoggerStartData
+        { logType
+        , ioMode = AppendMode
+        , handleLogger = makeMainLogger koreLogOptions
+        }
+    smtSolverLoggerStartData = LoggerStartData
+        { logType = maybe LogNone LogFileText smtSolverLogFile
+        , ioMode = WriteMode
+        , handleLogger = makeSmtSolverLogger
+        }
+
+makeSmtSolverLogger :: MonadIO m => Handle -> LogAction m SomeEntry
```

Comment:
I think the code to set up the loggers would be simpler if we replaced `makeSmtSolverLogger` and `makeMainLogger` with

```.hs
withSmtSolverLogger :: DebugSolverOptions -> (LogAction IO SomeEntry -> IO a) -> IO a

withMainLogger :: KoreLogOptions -> (LogAction IO SomeEntry -> IO a) -> IO a
```

`withLogger` becomes:

```.hs
withLogger
    koreLogOptions@KoreLogOptions { debugSolverOptions }
    continue
  =
    withMainLogger koreLogOptions $ \mainLogger ->
    withSmtSolverLogger debugSolverOptions $ \smtSolverLogger ->
        continue (mainLogger <> smtSolverLogger)
```

(`<>` for `LogAction` sends each `SomeEntry` to every logger.) I think this is easier to understand than `startLoggers` and we don't need to change `KoreLogType` or the command-line options.